mu-plugins/blocks/time/index.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Block Name: Time
+ * Description: Attempts to parse a time string like <code>[time]any-valid-time-string-here[/time]</code> and creates a format that shows it in the viewers local time zone.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\MU_Plugins\Time;
+
+use function WordPressdotorg\MU_Plugins\Helpers\register_assets_from_metadata;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type( __DIR__ . '/build' );
+	add_filter('the_content', __NAMESPACE__ . '\transform_time_blocks', 99, 1);
+}
+
+/**
+ * Builds the time block output.
+ *
+ * This implements replacing the raw time strings in the post content with formatted times able to be converted to local times with JS.
+ *
+ * @param string $content Post content.
+ * @return string Content with display times reformatted.
+ */
+function transform_time_blocks( $content) {
+	if ( empty( $content ) ) {
+		return $content;
+	}
+
+	// Find the time block elements by the classname "wporg-time"
+	$dom = new \DOMDocument();
+	$dom->loadHTML( $content );
+	$xpath = new \DOMXPath( $dom );
+	$time_elements = $xpath->query( "//*[contains(concat(' ', normalize-space(@class), ' '), ' wporg-time ')]" );
+	
+	foreach ( $time_elements as $time_element ) {
+		$time_content = $time_element->nodeValue;
+		$parsed_time = parse_time( $time_content );
+
+		if ( $parsed_time === null ) {
+			continue;
+		}
+		
+		// Build the link and abbr microformat.
+		$time_element->setAttribute( 'href', 'https://www.timeanddate.com/worldclock/fixedtime.html?iso=' . gmdate( 'Ymd\THi', $parsed_time ) );
+
+		$new_time_content = $dom->createElement( 'abbr', $time_content );
+		$new_time_content->setAttribute( 'class', 'wporg-time-date' );
+		$new_time_content->setAttribute( 'title', gmdate( 'c', $parsed_time ) );
```

Comment:
Do you know if there has been any feedback on the accessibility of this `abbr` + `title` approach?

Reply:
I don't, I'll ask

Reply:
I [asked](https://wordpress.slack.com/archives/C02RP4X03/p1689894967035259), feedback was that a `time` element would be better, and in particular the `title` on the `abbr` could be bad.